apps/dashboard/src/components/primitives/editor.tsx

```diff
@@ -114,19 +121,24 @@ type EditorProps = {
   placeholder?: string;
   className?: string;
   height?: string;
-  onChange: (val: string) => void;
-} & VariantProps<typeof editorVariants>;
+  onChange?: (val: string) => void;
+} & ReactCodeMirrorProps &
+  VariantProps<typeof editorVariants>;
 
-export const Editor = ({ value, placeholder, className, height, size, onChange }: EditorProps) => {
+export const Editor = ({
+  value,
+  placeholder,
+  className,
+  height,
+  size,
+  onChange,
+  extensions,
+  basicSetup,
+  ...restCodeMirrorProps
+}: EditorProps) => {
   const editor = useRef<HTMLDivElement>(null);
   const { setContainer } = useCodeMirror({
-    extensions: [
-      liquid({
-        variables: [{ type: 'variable', label: 'asdf' }],
-      }),
-      EditorView.lineWrapping,
-      baseTheme,
-    ],
```

Comment:
the language and extensions should not be hardcoded in the editor component as we want to use it for other languages as well